crates/solana-listener/src/component/signature_realtime_scanner.rs

```diff
@@ -20,16 +21,48 @@ use crate::component::log_processor::fetch_logs;
 use crate::component::signature_batch_scanner;
 use crate::{SolanaTransaction, TxStatus};
 
+/// Environment variable that expects a base58 encoded signature
+/// of the last processed signature we want to force.
+const FORCE_LAST_PROCESSED_SIGNATURE: &str = "FORCE_LAST_PROCESSED_SIGNATURE";
+
 #[tracing::instrument(skip_all, err, name = "realtime log ingestion")]
 pub(crate) async fn process_realtime_logs(
     config: crate::Config,
-    latest_processed_signature: Option<Signature>,
     rpc_client: Arc<RpcClient>,
     mut signature_sender: MessageSender,
+    state: impl SolanaListenerState,
 ) -> Result<(), eyre::Error> {
     let gateway_program_address = config.gateway_program_address;
     let gas_service_config_pda = config.gas_service_config_pda;
 
+    let force_last_processed_signature = std::env::var(FORCE_LAST_PROCESSED_SIGNATURE)?;
+    let latest_processed_signature = if force_last_processed_signature.is_empty() {
+        state.latest_processed_signature()
+    } else {
+        tracing::warn!(
+            "forcing last processed signature to {}",
+            force_last_processed_signature
+        );
+        Some(Signature::from_str(&force_last_processed_signature)?)
+    };
```

Comment:
nit: move this to the top-level `component`; to reduce the complexity of the realtime scanner fn

Reply:
fixed at 2a6d4801354c3457b66109aaf52209d87e37d4b2